examples/wasi.py

```diff
@@ -0,0 +1,22 @@
+from wasmer import Module, ImportObject, WasiVersion, WasiStateBuilder
+import os
+
+__dir__ = os.path.dirname(os.path.realpath(__file__))
+
+wasm_bytes = open(__dir__ + "/wasi.wasm", "rb").read()
+module = Module(wasm_bytes)
+
+wasi_state_builder = \
+    WasiStateBuilder("wasi_test_program"). \
+        argument("--test"). \
+        environment("COLOR", "true"). \
+        environment("APP_SHOULD_LOG", "false"). \
+        map_directory("the_host_current_dir", ".")
+
+import_object = module.generate_wasi_import_object(
+    wasi_state_builder,
+    module.wasi_version(False)
+)
```

Comment:
I believe the API for Python can be much simpler:
```suggestion
wasi = Wasi(
    args=["wasi_test_program", "--test"],
    env={"COLOR": "true", "APP_SHOULD_LOG": "false"},
    map_dir={"the_host_current_dir": "."}
)
import_object = wasi.get_imports_for_module(module)
```

Reply:
This also maps much closer to the API of `wasmer-js`, which makes it a bit easier to reason about.

Reply:
I've renamed `WasiStateBuilder` to `Wasi`. I've moved and renamed `generate_wasi_import_object` onto `Wasi` as `generate_import_object_for_module` (with an optional `version` argument).

However, notice that `Wasi` has the following other methods:
* `arguments`,
* `environments`,
* `preopen_directories`,
* `map_directories`.

Thus, the given example can also be written:

```py
wasi = \
    Wasi("wasi_test_program"). \
        arguments(["--test"]). \
        environments({"COLOR": "true", "APP_SHOULD_LOG": "false"}). \
        map_directories({"the_host_current_dir": "."})
```

I would like to keep this API as is, because it will allow to configure the Wasi File System more easily in the future for instance.

I can update the constructor to take optional `arguments`, `environments` and so on arguments however.

Reply:
Done, see https://github.com/wasmerio/python-ext-wasm/pull/167/commits/a55d655bde5d9c3217fa26e72674c5d34736ffa0. Commit message expanded here for the sake of simplicity:

> The following code:
> 
> ```py
> wasi = \
>     Wasi("wasi_test_program"). \
>         argument("--test"). \
>         environment("COLOR", "true"). \
>         environment("APP_SHOULD_LOG", "false"). \
>         map_directory("the_host_current_dir", ".")
> ```
> 
> could be rewritten:
> 
> ```py
> wasi = Wasi(
>     program_name="wasi_test_program",
>     arguments=["--test"],
>     environments={"COLOR": "true", "APP_SHOULD_LOG": "false"},
>     map_directories={"the_host_current_dir": "."}
> )
> ```
> 
> Both notations can be mixed.